app/assets/javascripts/discourse/app/components/modal/reorder-categories.js

```diff
@@ -1,24 +1,22 @@
-import Controller from "@ember/controller";
+import Component from "@ember/component";
+import { action } from "@ember/object";
 import { sort } from "@ember/object/computed";
-import Evented from "@ember/object/evented";
-import BufferedProxy from "ember-buffered-proxy/proxy";
+import { next } from "@ember/runloop";
+import { inject as service } from "@ember/service";
 import { ajax } from "discourse/lib/ajax";
 import { popupAjaxError } from "discourse/lib/ajax-error";
-import ModalFunctionality from "discourse/mixins/modal-functionality";
-import discourseComputed, { on } from "discourse-common/utils/decorators";
 
-export default Controller.extend(ModalFunctionality, Evented, {
-  init() {
-    this._super(...arguments);
-    this.categoriesSorting = ["position"];
-  },
+export default class ReorderCategories extends Component {
+  @service site;
+
+  categoriesSorting = ["position"];
 
-  @discourseComputed("site.categories.[]")
-  categoriesBuffered(categories) {
-    return (categories || []).map((c) => BufferedProxy.create({ content: c }));
-  },
+  @sort("site.categories", "categoriesSorting") categoriesOrdered;
 
-  categoriesOrdered: sort("categoriesBuffered", "categoriesSorting"),
+  init() {
+    super.init(...arguments);
+    next(() => this.reorder());
```

Comment:
Doing this on `next()` feels pretty weird... did you work out why it's required?

Reply:
Yeah, I spent a good portion of a day last week to refactor this component. It's our good friend mutation-after-read-in-component 🥲

We will have to eventually tackle that if we're going to rewrite it in glimmer, but for now the `next()` hack will do 🙂